src/components/App/DisplayPuzzleInstances/index.tsx

```diff
@@ -0,0 +1,137 @@
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+import styles from './DisplayPuzzleInstances.module.scss';
+import { PuzzleCustom } from '../../../types/api/puzzles/puzzle';
+import { getAllPuzzleInstances } from '../../../services';
+import { PuzzleInstance } from '@prisma/client';
+import { Button } from '../../Global';
+import { QRGenerator } from '../../App';
+import toast from 'react-hot-toast';
+
+type DisplayPuzzleInstancesProps = {
+  puzzlesList: PuzzleCustom[];
+};
+
+const DisplayPuzzleInstances = ({
+  puzzlesList
+}: DisplayPuzzleInstancesProps) => {
+  const [puzzleId, setPuzzleId] = useState<string>('');
+  const [puzzleInstances, setPuzzleInstances] = useState<PuzzleInstance[]>([]);
+  const [displayPuzzleInstances, setDisplayPuzzleInstances] = useState<
```

Comment:
I haven't read the rest of this file, but do you need 2 states to track lists of puzzle instances? If you do, then the variables should be named better

Reply:
One is for all the instances, one is for filtered instances. It was a bad name choice, I changed it!
`puzzleInstances` -> `allPuzzleInstances`
`displayPuzzleInstances` -> `puzzleInstances`

Reply:
Hope this make it readable

Reply:
Yep this helps!